crates/lib/src/qpu/api.rs

```diff
@@ -155,15 +150,10 @@ pub async fn retrieve_results(
 
     let request = GetControllerJobResultsRequest {
         job_execution_id: job_id.0,
-        target: Some(
-            execution_options
-                .connection_strategy
-                .get_results_target(quantum_processor_id)?,
-        ),
+        target: Some(execution_options.get_results_target(quantum_processor_id)?),
```

Comment:
Noting this because it's in the diff, even though it's preexisting; this is optional in the request, and so it should be optional from `get_results_target` as well.